### ISSUE TITLE
routine 1 : 2024.07.22~2024.08.31

### ISSUE THREAD

**Issue by lky473736** [lky473736/learning-AI#1]
2024.07.22. ~ 2024.08.31. 동안 아래와 같이 학습하여 AI를 체화시키자.

- **일주일에 3개 이상씩 UCI dataset 혹은 plotly에 있는 데이터를 하나 선정하여 classification or regression을 연습할 것**
   - 최소 100개의 records가 포함된 dataset
   - 모델은 최소 3개 이상
- **AI study에 적극적으로 참여**
   - 매주 세미나 내용을 정리 후 이해된 내용을 write
   - 자료 잘 숙지하여 ipynb로 정리 및 코드로 classification
   - https://wikidocs.net/book/2788 참고하여 pytorch ML 입문 부분 학습

- **<혼자 공부하는 머신러닝과 딥러닝> 책 3회독** 후 앤드류 응 머신러닝 강좌 학습하기 (2학기 계획, https://www.coursera.org/specializations/machine-learning-introduction?&utm_source=gg&utm_medium=sem&utm_campaign=b2c_apac_machine-learning-introduction_stanford-deeplearning.ai_ftcof_specializations_arte_mar-24_dr_geo-set-1-multi_pmax_gads_lg-all&utm_content=b2c&campaignid=21113902254&adgroupid=&device=c&keyword=&matchtype=&network=x&devicemodel=&adpostion=&creativeid=&hide_mobile_promo&gad_source=1&gclid=CjwKCAjwhvi0BhA4EiwAX25uj9sHoC9TY8_VdKnCyIWGBO36wPHSc0NyuMUZxff_VxXY6ncvEj_dZBoCxfgQAvD_BwE)

**Comment by lky473736** [comment]
- 2024.07.26.
    - 자료 읽기
        - linear regression with stochastic gradient regression : https://towardsdatascience.com/step-by-step-tutorial-on-linear-regression-with-stochastic-gradient-descent-1d35b088a843?gi=c14b6ca02405
        - 머신 러닝 핵심 알고리즘 (pdf) : KNN, SVM, decision tree

**Comment by lky473736** [comment]
- 2024.07.27.
    - 자료 읽기
        - support vector machine : https://m.blog.naver.com/winddori2002/221662413641 
        - decision tree : https://lucy-the-marketer.kr/ko/growth/decision-tree-and-impurity/
    - 머신러닝 이해 안되는 부분 다시 학습
        - 앙상블 학습 (ramdom forest) **<-- 내일 진행으로 변경**
    - 유리 데이터를 이용한 classification, regression을 진행
        -  data : https://archive.ics.uci.edu/dataset/42/glass+identification
        -  혼공머신 머신러닝 챕터 복습 겸, 이번 주말엔 2개의 dataset을 이용하여 책에 나와있는 순서 그대로 학습을 진행해본다.
        -  순서는 (내 기억 상) 아래와 같을 것, 밤을 새야할 것 같으니 흥미진진하다.
            - KNN
            - KNR, linear regression, polynormial regression
            - ridge, lasso
            - logistic regression
            - SGD (stochastic gradient descent) 
            - decision tree
            - grid search, random search
            - random forest
            - 아래는 비지도 학습, 다른 부분처럼 자세히 다루지는 않고, 그냥 API를 사용하는 정도로 연습
                - K-means
                - PCA

**Comment by lky473736** [comment]
- 2024.07.28.
    - 자료 읽기
        - logistic regression 
              - https://ratsgo.github.io/machine%20learning/2017/04/02/logistic/
              - https://m.blog.naver.com/winddori2002/221706766540
        - random forest  : https://velog.io/@dlskawns/Machine-Learning-Random-Forest-정리-구성원리-파악-및-모델-작성
    - 머신러닝 이해 안되는 부분 다시 학습
        - 앙상블 학습 (ramdom forest) 
    - 유리 데이터를 이용한 classification, regression을 진행
        - 새벽에 EDA, KNN, SVM, logistic regression, decision tree까지 하기
        - 자료를 읽고, random forest, K-means 및 나머지 regression 진행
            - regression 진행 시 feature selection한 feature만 가지고 진행하기
            - score 안좋으면 poly

**Comment by lky473736** [comment]
- 2024.07.29.
    - 자료 읽기
         - PCA
             -  https://ddongwon.tistory.com/114
         - 딥러닝의 원리, 종류별 원리
             - https://www.ibm.com/kr-ko/topics/deep-learning
  ~~- 혼공머신 딥러닝편 340p~391p 및 정리~~
  ~~- 일전에 했었던 report들에 DL 코드 추가하기 (1)~~
          ~~- raisin classification, regression에 keras로 진행한 코드 추가 및 기존 ML과 성능 비교~~
     ~~- ADsP~~
         ~~- 2시간 특강 1회독 : https://www.youtube.com/watch?v=6h6Dsh6IMgA~~ **<--내일 진행**

   - 혼공머신 머신러닝 ensemble, PCA 정리 후 코드 실습
   - <수학> 연습문제, 기본문제 체크

**Comment by lky473736** [comment]
- 2024.07.30
  - 자료 읽기
    - 빅데이터분석개론(2-여름) 딥러닝 파트 자료 (입문 수준)
      - https://jiwoncho20213135python.tistory.com/15
      - ppt

  - (어제 못한 거) 혼공머신 딥러닝편 340p~420p 공부 및 정리
  - 일전에 했었던 report들에 DL 코드 추가하기 (1)
       - raisin classification에 keras로 진행한 코드 추가 및 기존 ML과 성능 비교

**Comment by lky473736** [comment]
- 2024.07.31.
  - 혼공머신 딥러닝편 422p~463p
  - raisin classification에 CNN 추가하기 
  - refactoring
       - report 폴더에 ML, DL로 문서 seperate
           - ML : EDA, ML
           - DL : dense, CNN, RNN, LSTM...

**Comment by lky473736** [comment]
- 2024.08.01
  - DL_classification_raisin에서 output layer의 node 갯수가 1이여야 오류가 안나는 상황을 해결
  - 여태껏 했던 모든 classification, regression에 dense layer struction 시도 (1)
      - glass classification, regression
      - diabetes classification, regression
  - problem solving
      - https://www.acmicpc.net/problem/2405
      - https://www.acmicpc.net/problem/22940

**Comment by lky473736** [comment]
- 2024.08.04.
    - HARTH classification DL 마무리짓기
    - glass regression DL 
    - ADsP 총정리 회독
    - 자료 읽기
        - PCA 원리 복습 : https://ddongwon.tistory.com/114
        - 머신러닝과 딥러닝의 진짜 의미 : https://tensorflow.blog/%EC%BC%80%EB%9D%BC%EC%8A%A4-%EB%94%A5%EB%9F%AC%EB%8B%9D/1-%EB%94%A5%EB%9F%AC%EB%8B%9D%EC%9D%B4%EB%9E%80-%EB%AC%B4%EC%97%87%EC%9D%B8%EA%B0%80/

**Comment by lky473736** [comment]
- 2024.08.05.
    - glass classification DL
    - diabetes classification DL 마무리짓기
    - ADsP 총정리 회독 2
    - 자료 읽기
        - https://machinelearningmastery.com/how-to-develop-convolutional-neural-network-models-for-time-series-forecasting/
        -  https://velog.io/@not12m10/Convolution%EC%9D%98-%EA%B0%9C%EB%85%90%EA%B3%BC-%EC%97%B0%EC%82%B0%EC%97%90-%EB%8C%80%ED%95%98%EC%97%AC

- **이번 주 계획**
    - 주말부터 전에 들었던 딥러닝 강좌 다시 수강 시작하기 (내용 다 까먹음)
    - CNN에 대한 완전한 이해를 기반으로 새 데이터에 대해 분류, regression 진행해보기
        - breast cancer : ML regression, ML classification, DL regression, DL classification
        - rice : ML regression, ML classification, DL regression, DL classification
    - 일단 ADsP 시험에 집중, 토요일날 시험 끝나면 RNN 제대로

**Comment by lky473736** [comment]
- 2024.08.06.
    - glass classification CNN
    - diabetes classification CNN
    - 자료 읽기 
         - https://machinelearningmastery.com/how-to-develop-convolutional-neural-network-models-for-time-series-forecasting/
         - https://velog.io/@not12m10/Convolution%EC%9D%98-%EA%B0%9C%EB%85%90%EA%B3%BC-%EC%97%B0%EC%82%B0%EC%97%90-%EB%8C%80%ED%95%98%EC%97%AC
     - 혼공머신 CNN-3
     - ADsP 1, 2과목 빠르게 훑고 마무리 암기

**Comment by lky473736** [comment]
- 2024.08.07. ~ 2024.08.08.
    - 혼공머신 RNN-1
    - CNN에 대한 제대로 된 이해 
	  - split-sequence
	  - conv1d, conv2d, conv3d
    - breast cancer ML

**Comment by lky473736** [comment]
- 2024.08.09.
    - 혼공머신 RNN-2
    - 자료 읽기
         - 08장 심층신경망
         - 09장 컨볼루션 신경망
         - 11장 순환신경망
         - LSTM, GRU에 대한 단계적 설명
     - breast cancer DL

**Comment by lky473736** [comment]
- 2024.08.10. ~ 08.13.
    - 혼공머신 RNN-3, prev chapter 1
    - 자료 읽기
         - transformer 알아보기
         - transformer의 큰 그림 이해 : 기술적 복잡함 없이 핵심 아이디어 파악하기
     - vehicle DL, ML (원래 rice였다가 너무 쉬워보여서 다른걸로 변경, regression 진행 예정)
     - california housing value DL, ML
     - (08/10 ~ 08/13) SGD를 모방하여 딥러닝 학습을 직접 구현해보기 (미분, back propagation)

**Comment by lky473736** [comment]
https://machinelearningmastery.com/how-to-develop-rnn-models-for-human-activity-recognition-time-series-classification/

**Comment by lky473736** [comment]
- 2024.08.17. ~ 2024.08.18.
    - FAQ 스스로 작성
    - 인터넷 강좌
         - 4강~7강 : https://www.youtube.com/playlist?list=PLWtr7MRpQi5ATPYo3xN21Bwwd_Ut4rY0d (앞으로 하루에 4강씩 들으면서 ai 기본 복습하기)
         - summary 정리 : principle of machine learning and deep learning - stf101 
     - vehicle DL
     - stock DL (rnn, cnn, lstm)
     - 1D-CNN : U-net
     - 자료 읽기
         - [RNN] https://towardsdatascience.com/deep-learning-illustrated-part-4-recurrent-neural-networks-d0121f27bc74 
         - [LSTM] https://medium.com/@myskill.id/time-series-prediction-lstm-759763728c48, https://youtu.be/8HyCNIVRbSU?si=CLg7RGUgV2i-y_Sg 
         - [Transformer] https://medium.com/@actionpower/%ED%8A%B8%EB%9E%9C%EC%8A%A4%ED%8F%AC%EB%A8%B8-transformer-%EC%95%8C%EC%95%84%EB%B3%B4%EA%B8%B0-1d595a208230

**Comment by lky473736** [comment]
2024.08.19. 부터 개강까지의 목표
- 그동안 했었던 AI study 혹은 seminar에서의 블로그 자료 정독 
- hands-on machine learning 일부 내용 예습

**Comment by lky473736** [comment]
- 2024.08.23. ~ 2024.08.25.
    - FAQ 다시 상세히 작성해보기
    - U-net 이해하기
        - HARTH
        - DL ion switching
        - ML ion switching
    - 그동안 밀렸던 블로그 자료 전부 쭉 읽기
    - samsung prediction
    - pixel war idea 생각해보기
